runner_manager/main.py

```diff
@@ -1,13 +1,39 @@
 import logging
-
+from fastapi import Depends, FastAPI, HTTPException, Response, Security, status
+from fastapi.security import APIKeyHeader, APIKeyQuery
 from fastapi import FastAPI, Response
 
-from runner_manager.dependencies import get_queue
+from runner_manager.dependencies import get_queue, get_settings
 from runner_manager.jobs.startup import startup
 
 log = logging.getLogger(__name__)
 
 app = FastAPI()
+api_key_query = APIKeyQuery(name="api-key", auto_error=False)
+api_key_header = APIKeyHeader(name="x-api-key", auto_error=False)
+
+def get_api_key(
+    api_key_query: str = Security(api_key_query),
+    api_key_header: str = Security(api_key_header),
+) -> str:
+    settings = get_settings()
+    if not settings.api_key:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="API Key not configured in settings",
+        )
```

Comment:
If we haven't configured the api key then we can proceed with the request without making any kind of authentication.

```suggestion
    if not settings.api_key:
        return settings.api_key
```